clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -45,9 +45,19 @@ import           Clash.Netlist.BlackBox.Types   (HdlSyn (..))
 --
 data Binding a = Binding
   { bindingId :: Id
+    -- ^ The core identifier for this binding.
   , bindingLoc :: SrcSpan
+    -- ^ The source location of this binding in the original source code.
   , bindingSpec :: InlineSpec
+    -- ^ the inline specification for this binding, in the original source code.
+  , bindingIsPrim :: Bool
```

Comment:
Would you be opposed to make this `IsPrimitive` to avoid boolean blindness?

Reply:
I would not, one moment...